matrix-sdk-android/src/test/java/org/matrix/android/sdk/internal/session/room/aggregation/livelocation/LiveLocationAggregationProcessorTest.kt

```diff
@@ -0,0 +1,387 @@
+                isLocalEcho = false
+        )
+        val resultEmptySenderId = liveLocationAggregationProcessor.handleBeaconLocationData(
+                realm = fakeRealm.instance,
+                event = eventEmptySenderId,
+                content = beaconLocationData,
+                roomId = A_ROOM_ID,
+                relatedEventId = AN_EVENT_ID,
+                isLocalEcho = false
+        )
+
+        resultNoSenderId shouldBeEqualTo false
+        resultEmptySenderId shouldBeEqualTo false
+    }
+
+    @Test
+    fun `given beacon location data when location is less recent than the saved one then it is ignored`() {
+        val event = Event(eventId = AN_EVENT_ID, senderId = A_SENDER_ID)
+        val beaconLocationData = MessageBeaconLocationDataContent(
+                unstableTimestampMillis = A_TIMESTAMP - 60_000
+        )
+        val lastBeaconLocationContent = MessageBeaconLocationDataContent(
+                unstableTimestampMillis = A_TIMESTAMP
+        )
+        mockLiveLocationShareAggregatedSummaryEntityForEvent(lastBeaconLocationContent = lastBeaconLocationContent)
+
+        val result = liveLocationAggregationProcessor.handleBeaconLocationData(
+                realm = fakeRealm.instance,
+                event = event,
+                content = beaconLocationData,
+                roomId = A_ROOM_ID,
+                relatedEventId = AN_EVENT_ID,
+                isLocalEcho = false
+        )
+
+        result shouldBeEqualTo false
+    }
+
+    @Test
+    fun `given beacon location data when location is more recent than the saved one then it is aggregated`() {
+        val event = Event(eventId = AN_EVENT_ID, senderId = A_SENDER_ID)
+        val locationInfo = LocationInfo(geoUri = A_GEO_URI)
+        val beaconLocationData = MessageBeaconLocationDataContent(
+                unstableTimestampMillis = A_TIMESTAMP,
+                unstableLocationInfo = locationInfo
+        )
+        val lastBeaconLocationContent = MessageBeaconLocationDataContent(
+                unstableTimestampMillis = A_TIMESTAMP - 60_000
+        )
+        val entity = mockLiveLocationShareAggregatedSummaryEntityForEvent(lastBeaconLocationContent = lastBeaconLocationContent)
+
+        val result = liveLocationAggregationProcessor.handleBeaconLocationData(
+                realm = fakeRealm.instance,
+                event = event,
+                content = beaconLocationData,
+                roomId = A_ROOM_ID,
+                relatedEventId = AN_EVENT_ID,
+                isLocalEcho = false
+        )
+
+        result shouldBeEqualTo true
+        val savedLocationData = ContentMapper.map(entity.lastLocationContent).toModel<MessageBeaconLocationDataContent>()
+        savedLocationData?.getBestTimestampMillis() shouldBeEqualTo A_TIMESTAMP
+        savedLocationData?.getBestLocationInfo()?.geoUri shouldBeEqualTo A_GEO_URI
+    }
+
+    private fun mockLiveLocationShareAggregatedSummaryEntityForEvent(
```

Comment:
it could be helpful to name these helpers with the data they're fetching 

```
givenLastRoomBeaconQueryReturns(roomId, eventId, beaconLocation)
```

```kotlin
val activeSummary = LiveLocationShareAggregatedSummaryEntity(
  eventId = "ignored-value",
  roomId = A_ROOM_ID,
  userId = A_SENDER_ID,
  isActive = true
)
givenLocationSummaryQueryReturns(activeSummary)
```